src/server/session.ts

```diff
@@ -504,18 +504,18 @@ namespace ts.server {
         // If `getResultsForPosition` returns results for a project, they go in here
         const resultsMap = new Map<Project, readonly TResult[]>();
 
-        const queue: ProjectAndLocation[] = [];
+        const queue = createQueue<ProjectAndLocation>();

```

Comment:
Even here, does order matter if not we could get away with having to move array elements right?

Reply:
I believe order does matter, though that doesn't necessarily mean we couldn't accomplish the same thing with a stack.

Reply:
why does order matter ? in all find all references etc when aggregating results between project, the order shouldnt matter right ? Yes our baselines may change but from end to end result perspective order shouldnt matter ?

Reply:
I believe the search order affects the value of `isDefinition`.  I believe it's also the case that FAR is asymmetric (B being a reference to A, does not imply that A is a reference to B), but I think we've already decided to ignore that restriction, so it's probably just `isDefinition`.  Now that `isDefinition` is computed in a post-pass, it's possible that we can drop the strict ordering, but I'm frankly pretty burned out on rewriting this code.  I have no objections if you'd like to, but I'd suggest we wait until 4.9, now that we're in the RC period.